### PR TITLE
Update `Array.hx` to have `set` and `get` methods

### PR DESCRIPTION
**Pull request by Just-Feeshy**
Example of how it works:

Main.hx:
```haxe
package;

class Main {
		public static function main() {
		    var a = new cxx.std.Array<Int, 5>();
		    a.set(0, 5);
			a.set(1, 3);
			a.set(2, 1);

			trace("Hello, World!" + (a.get(0) + a.get(1) + a.get(2)));
		}
}
```

To Main.cpp:
```cpp
#include "Main.h"

#include <array>
#include <memory>
#include <string>
#include "_AnonStructs.h"
#include "haxe_Log.h"

using namespace std::string_literals;

void Main::main() {
	std::shared_ptr<std::array<int, 5>> a = std::make_shared<std::array<int, 5>>();

	(*a)[0] = 5;
	(*a)[1] = 3;
	(*a)[2] = 1;
	haxe::Log::trace("Hello, World!"s + std::to_string((((*a)[0]) + ((*a)[1]) + ((*a)[2]))), haxe::shared_anon<haxe::PosInfos>("Main"s, "Main.hx"s, 10, "main"s));
}
```

Output:
```bash
Main.hx:10: Hello, World!9
```

### CODE REVIEW

File: std/cxx/std/Array.hx [hunk 1]
```diff
@@ -17,4 +17,12 @@ extern class Array<T, @:const Count: Int>  {
 	public function maxSize(): cxx.num.SizeT;
 
 	public function fill(value: T): Void;
+
+	@:arrayAccess
+        @:nativeFunctionCode("((*{this})[{arg0}])")
```
Comment (msteinbeck) [inline std/cxx/std/Array.hx:22]: I propose to map this to `at`: https://en.cppreference.com/w/cpp/container/array/at

File: std/cxx/std/Array.hx [hunk 2]
```diff
@@ -17,4 +17,12 @@ extern class Array<T, @:const Count: Int>  {
...
+	public function get(index: Int): T;
+
+	@:arrayAccess
+        @:nativeFunctionCode("(*{this})[{arg0}] = {arg1}")
```
Comment (msteinbeck) [inline std/cxx/std/Array.hx:26]: I propose to map this to at: https://en.cppreference.com/w/cpp/container/array/at

File: std/cxx/std/Array.hx (hunk 1 above)
Comment (Just-Feeshy) [reply std/cxx/std/Array.hx:22]: You've got a good point.

Comment (Just-Feeshy) [reply std/cxx/std/Array.hx:22]: However, I sometimes don't want to waste time on bound checking if I don't need to.

**Comment by Just-Feeshy** [comment]
I added a warning comment to let others know that these functions aren't safe.

**Pull request merged**